src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -583,7 +583,12 @@ def send_raw_request(cli_ctx, method, uri, headers=None, uri_parameters=None,  #
     uri_parameters = result or None
 
     if '://' not in uri:
+        # Default to Azure Resource Manager
         uri = cli_ctx.cloud.endpoints.resource_manager + uri.lstrip('/')
+        # Use configured ARM's resource ID, following the same behavior as
+        # azure.cli.core.commands.client_factory._get_mgmt_service_client
+        resource = resource or cli_ctx.cloud.endpoints.active_directory_resource_id
```

Comment:
What kind of scene is `'://' not in uri`? `resource = resource or cli_ctx.cloud.endpoints.active_directory_resource_id` is it expected to work only in this scenario?

Reply:
When the user doesn't provide the full URL `https://...` but only the resource ID `/subscription/...`, `az rest` prefixes the ID with `resource_manager` endpoint and fills `resource`.

Reply:
I have refactored the code - now `resource` will be set to `https://management.core.windows.net/` when customer uses either form:

```
--uri /subscriptions/01/resourcegroups/02?api-version=2019-07-01
--uri https://management.azure.com/subscriptions/01/resourcegroups/02?api-version=2019-07-01
```

Corresponding tests are also added.